src/atlas/grid/detail/grid/Gaussian.cc

```diff
@@ -246,6 +246,28 @@ StructuredGrid::grid_t* reduced_gaussian( const std::vector<int>& nx, const Doma
     return new StructuredGrid::grid_t( xspace( _nx ), Spacing( yspace ), Projection(), domain );
 }
 
+StructuredGrid::grid_t* reduced_gaussian( const std::vector<int> & nx, double centre[], double stretch) {
```

Comment:
function could be called `rotated_stretched_reduced_gaussian` and should take a `vector<idx_t>` instead of `vector<int>` for `nx`. That is also the return type of `idx_vector` which you used.